### PR TITLE
Api 2.0 build docs website

### PR DESCRIPTION
**Pull request by chentong7**
## Description

> Rendering multi API document version on FluidFramework public site. Retrieve API package content for dropdown selected version number. If the version number is not selected, it will show API overview page.

## Sample

https://github.com/microsoft/FluidFramework/assets/114451900/6d4be4f2-2e67-48c2-9bed-3e92cacb7289

AB#6175

### CODE REVIEW

File: docs/api-markdown-documenter/render-api-documentation.js [hunk 1]
```diff
@@ -53,7 +66,7 @@ async function renderApiDocumentation() {
 			ApiItemKind.Namespace,
 		],
 		newlineKind: "lf",
-		uriRoot: "/docs/apis",
+		uriRoot: "/docs/apis/" + apiVersionNum,
```
Comment (Josmithr) [inline docs/api-markdown-documenter/render-api-documentation.js:69]:
```suggestion
		uriRoot: `/docs/apis/`${apiVersionNum}`,
```

File: docs/api-markdown-documenter/render-api-documentation.js [hunk 2]
```diff
@@ -21,10 +21,23 @@ const { buildNavBar } = require("./build-api-nav");
 const { renderAlertNode, renderBlockQuoteNode, renderTableNode } = require("./custom-renderers");
 const { createHugoFrontMatter } = require("./front-matter");
 
-const apiReportsDirectoryPath = path.resolve(__dirname, "..", "_api-extractor-temp", "_build");
-const apiDocsDirectoryPath = path.resolve(__dirname, "..", "content", "docs", "apis");
+async function renderApiDocumentation(apiVersionNum) {
```
Comment (Josmithr) [inline docs/api-markdown-documenter/render-api-documentation.js:24]: Would you mind adding some documentation here explaining how `apiVersionNum` is used? I just want to make sure the contract is clear for future readers :)

File: docs/api-markdown-documenter/index.js [hunk 3]
```diff
@@ -4,16 +4,33 @@
...
+	versions = yaml.load(fs.readFileSync("./data/versions.yaml", "utf8"));
+} catch (e) {
+	console.log(e);
+}
+
+docVersions = renderMultiVersion
+	? versions.params.previousVersions
+	: versions.params.currentVersion;
+
+docVersions.forEach((version) => {
```
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:25]: We should probably use `Promise.all` here, rather than running each promise in sequence.

File: docs/api-markdown-documenter/index.js [hunk 4]
```diff
@@ -4,16 +4,33 @@
...
-		console.error("API docs could not be written due to an error:", error);
-		process.exit(1);
-	},
-);
+const renderMultiVersion = process.argv[2];
+
+let versions;
+
+try {
+	versions = yaml.load(fs.readFileSync("./data/versions.yaml", "utf8"));
```
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:16]: I believe @zhenmichael's PR is updating this to be a `json` file, which should make your life easier here :)

File: packages/dds/migration-shim/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# @fluid-experimental/migration-shim
```
Comment (Josmithr) [inline packages/dds/migration-shim/README.md:1]: This looks like an unrelated change. Probably from a merge. This package was deleted, so you can just delete this file.

File: docs/data/versions.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+params:
```
Comment (Josmithr) [inline docs/data/versions.yaml:1]: Before we check in, I believe we'll want to make sure 2.0 is removed from this list. I don't believe we want to turn on our 2.0 docs until we're ready to launch the beta (we're actively working on trimming the API surface for it between now and release).

I think we just want the 1.0 docs for now.

@rohandubal @sashasimic interested in your opinions on this :)

File: docs/themes/thxvscode/layouts/partials/apiNav.html [hunk 7]
```diff
@@ -25,13 +25,16 @@
 {{ end }}
 
 {{ $section := .CurrentSection }}
-{{ $api_section := .Site.GetPage "/docs/apis/_index.md" }}
+{{ $api_section := .Site.GetPage "/docs/apis/1.0/_index.md" }}
```
Comment (Josmithr) [inline docs/themes/thxvscode/layouts/partials/apiNav.html:28]: We'll want to make this dynamic. Rather than having 2 hard-coded sections for the different versions, we want to check the `versions.json` file to see which ones are enabled, and load those.

File: docs/themes/thxvscode/layouts/partials/apiNav.html [hunk 8]
```diff
@@ -80,9 +83,16 @@
...
     <hr/>
     <ul class="nav" id="secondary-nav">
-        <li {{ if $in_api_section }} class="active" {{ end }}>
-            <a class="docs-home" href="/docs/apis/" {{ if $in_api_section }}
-                aria-label="Current Page: API Overview" {{ end }}>API Overview</a>
+        <li>
+            <select name="API version" id="api-version" onchange="window.location.href = this.value;">
+                {{ $selectedApiVersion := trim $current.RelPermalink "/" }}
+                {{ $selectedApiVersion = index (split $selectedApiVersion "/") 2 }}
+                {{ $isV2 := eq $selectedApiVersion "2.0"}}
```
Comment (Josmithr) [inline docs/themes/thxvscode/layouts/partials/apiNav.html:90]: Same as my comment [above](https://github.com/microsoft/FluidFramework/pull/18710/files#r1421097271) - we need this logic to be dynamic based on the contents of `versions.json`. Otherwise, we'll have to modify this code every time we release a new version or stop supporting an old version. We'd like to control that strictly from the versions file.

**Review by Josmithr — COMMENTED** [review]
Looking good! My main feedback is just that we need to not have hard-coded values in the hugo code for 1.0 and 2.0 - the version options we display should be based on the contents of the `versions.json` file (currently `versions.yaml` on this branch, but I believe @zhenmichael has changed this to a json file on his branch). Otherwise we'll have to modify the Hugo layout code any time our supported set of versions changes, which we don't want.

File: docs/data/versions.yaml (hunk 6 above)
Comment (rohandubal) [reply docs/data/versions.yaml:1]: Agreed, we should work to stage this change (if possible) or test in an alternate manner and keep a separate PR closer to the launch.

Comment (chentong7) [reply docs/data/versions.yaml:1]: Understood. Remove 2.0 for now

File: docs/api-markdown-documenter/render-api-documentation.js (hunk 2 above)
Comment (chentong7) [reply docs/api-markdown-documenter/render-api-documentation.js:24]: Make sense! Adding the comments here!

File: docs/data/versions.yaml (hunk 6 above)
Comment (sashasimic) [reply docs/data/versions.yaml]: Agreed. We can toggle it when 2.0 rc is released.

File: docs/api-markdown-documenter/index.js (hunk 4 above)
Comment (chentong7) [reply docs/api-markdown-documenter/index.js]: Updated accordingly.

File: docs/data/versions.yaml (hunk 6 above)
Comment (chentong7) [reply docs/data/versions.yaml]: Updated. Seems like index.js takes care of multi-version flag. It's off now

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 7 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html]: Updated

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 8 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html]: Up-to-date

File: docs/api-markdown-documenter/index.js [hunk 9]
```diff
@@ -4,16 +4,30 @@
...
-	() => {
-		console.log(chalk.green("API docs written!"));
-		process.exit(0);
-	},
-	(error) => {
-		console.error("API docs could not be written due to an error:", error);
-		process.exit(1);
-	},
-);
+const renderMultiVersion = process.argv[2];
```
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:10]: Let's add some top-level documentation here explaining what this script does and, in particular, what arguments are expected and what they mean.

File: docs/api-markdown-documenter/index.js [hunk 10]
```diff
@@ -4,16 +4,30 @@
...
+				process.exit(0);
+			},
+			(error) => {
+				console.error(`API ${version} docs could not be written due to an error:`, error);
+				process.exit(1);
+			},
+		),
+	);
+});
+Promise.all(apiDocRenders);
```
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:33]: This needs to be awaited.
```suggestion
await Promise.all(apiDocRenders);
```

File: docs/api-markdown-documenter/index.js [hunk 11]
```diff
@@ -4,16 +4,30 @@
...
+
+docVersions.forEach((version) => {
+	apiDocRenders.push(
+		renderApiDocumentation(version).then(
+			() => {
+				console.log(chalk.green(`API ${version} docs written!`));
+				process.exit(0);
+			},
+			(error) => {
+				console.error(`API ${version} docs could not be written due to an error:`, error);
```
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:27]: Thanks for adding the version to the logs here! That will make debugging a lot easier!

File: docs/api-markdown-documenter/render-api-documentation.js [hunk 12]
```diff
@@ -21,10 +21,21 @@ const { buildNavBar } = require("./build-api-nav");
...
 const { createHugoFrontMatter } = require("./front-matter");
 
-const apiReportsDirectoryPath = path.resolve(__dirname, "..", "_api-extractor-temp", "_build");
-const apiDocsDirectoryPath = path.resolve(__dirname, "..", "content", "docs", "apis");
+/**
+ * To rendering multiple versions of the API docs, this function is called for each version.
+ * @param apiVersionNum api version number, e.g. "2.0"
+ */
+async function renderApiDocumentation(apiVersionNum) {
+	const apiReportsDirectoryPath = path.resolve(__dirname, "..", "_api-extractor-temp", "_build");
```
Comment (Josmithr) [inline docs/api-markdown-documenter/render-api-documentation.js:29]: This needs to read from the directory for the corresponding version too, right? Otherwise we'll generate the same versions docs for each output directory, ignoring that version number we were given.

File: docs/api-markdown-documenter/render-api-documentation.js [hunk 13]
```diff
@@ -21,10 +21,21 @@ const { buildNavBar } = require("./build-api-nav");
 const { renderAlertNode, renderBlockQuoteNode, renderTableNode } = require("./custom-renderers");
 const { createHugoFrontMatter } = require("./front-matter");
 
-const apiReportsDirectoryPath = path.resolve(__dirname, "..", "_api-extractor-temp", "_build");
-const apiDocsDirectoryPath = path.resolve(__dirname, "..", "content", "docs", "apis");
+/**
+ * To rendering multiple versions of the API docs, this function is called for each version.
+ * @param apiVersionNum api version number, e.g. "2.0"
```
Comment (Josmithr) [inline docs/api-markdown-documenter/render-api-documentation.js:26]: Nit:
```suggestion
 * @param {number} apiVersionNum - The release version for which API docs will be rendered.
```

It would also probably be good to call out how we organize the docs artifacts currently, since this functionality depends on that. E.g. `2.0` means we will read from `../api-extractor-temp/_build/2.0` and we will write to `content/docs/apis/2.0`.

File: docs/data/versions.json [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+		"currentVersion": ["2.0"],
+		"ltsVersion": ["1.0"],
+		"previousVersions": ["1.0"]
+	},
+	"apiDocContentPath": {
+		"Select version": "/docs/apis/",
+		"Fluid Framework 1.0": "/docs/apis/1.0/",
+		"Fluid Framework 2.0 Alpha": "/docs/apis/2.0/"
+	},
+	"apiDocIndexPath": ["/docs/apis/1.0/_index.md", "/docs/apis/2.0/_index.md"]
```
Comment (Josmithr) [inline docs/data/versions.json:12]: I don't think we need to have this extra metadata (apiDocsContentPath and apiDocsIndexPath) for the paths here. We can always derive `docs/apis/<version>` from the version number we're looking at.

Comment (Josmithr) [reply docs/data/versions.json:12]: If it's useful (and it might be), we could add an `apiDocsDirectory` (or something) value that just points to `docs/apis`. Then we could use that plus the version numbers to derive the paths you need, without having to hard-code `docs/apis` all over the place in the Hugo code.

File: docs/themes/thxvscode/layouts/partials/apiNav.html [hunk 15]
```diff
@@ -24,12 +24,16 @@
     {{ end }}
 {{ end }}
 
-{{ $section := .CurrentSection }}
-{{ $api_section := .Site.GetPage "/docs/apis/_index.md" }}
-{{ $in_api_section := (.IsDescendant $api_section) }}
+{{ $selectedApiVersion := ""}}
```
Comment (Josmithr) [inline docs/themes/thxvscode/layouts/partials/apiNav.html:27]: As we discussed in the meeting yesterday, this should probably default to `currentVersion` from `versions.json`.

We can make sure we have the nice landing pages we need by copying what is currently in `docs/_replacements/docs/apis` into `docs/_replacements/docs/apis/1.0` and `docs/_replacements/docs/apis/2.0`.

File: docs/themes/thxvscode/layouts/partials/apiNav.html [hunk 16]
```diff
@@ -80,9 +84,15 @@
...
     <ul class="nav" id="secondary-nav">
-        <li {{ if $in_api_section }} class="active" {{ end }}>
-            <a class="docs-home" href="/docs/apis/" {{ if $in_api_section }}
-                aria-label="Current Page: API Overview" {{ end }}>API Overview</a>
+        <li>
+            <select name="API version" id="api-version" onchange="window.location.href = this.value;">
+                {{ $selectedApiVersion := trim $current.RelPermalink "/" }}
+                {{ $selectedApiVersion = index (split $selectedApiVersion "/") 2 }}
+                {{ range $version, $path := $.Site.Data.versions.apiDocContentPath }}
+                    {{ $isCurrentVersion := strings.Contains $version $selectedApiVersion }}
```
Comment (Josmithr) [inline docs/themes/thxvscode/layouts/partials/apiNav.html:92]: Can this not be an exact equality check? Curious if `contains` is necessary. If so, it would be good to add a comment explaining why.

File: docs/themes/thxvscode/layouts/partials/apiNav.html [hunk 17]
```diff
@@ -80,9 +84,15 @@
     </ul>
     <hr/>
     <ul class="nav" id="secondary-nav">
-        <li {{ if $in_api_section }} class="active" {{ end }}>
-            <a class="docs-home" href="/docs/apis/" {{ if $in_api_section }}
-                aria-label="Current Page: API Overview" {{ end }}>API Overview</a>
+        <li>
+            <select name="API version" id="api-version" onchange="window.location.href = this.value;">
+                {{ $selectedApiVersion := trim $current.RelPermalink "/" }}
+                {{ $selectedApiVersion = index (split $selectedApiVersion "/") 2 }}
```
Comment (Josmithr) [inline docs/themes/thxvscode/layouts/partials/apiNav.html:90]: Nit: we do this same splitting logic in a couple of places. Is there any way for us to extract this into a sharable helper of some kind? I don't know how easy this is in Hugo, so it may not be worth the effort if it's complicated to do.

At the very least, we should document what we expect the URL to look like and what the magic number `2` here is for.

**Review by Josmithr — COMMENTED** [review]
Looking closer, but there are still a couple of issues we need to resolve. I left comments for them, but at a high level:
1. We need to make sure the API metadata we're *reading in* when generating the API docs is coming from the specified version. Right now it looks like we're generating the docs from the same source directory regardless of the version being specified.
2. We should default the API docs version selection to whatever is specified under `version.json`'s `currentVersion`. We can create nice landing pages for each of the versions via `docs/_replacements`. Once you've added those, we'll want to make a few changes to them to highlight the right packages for each version.

The packages we want to light up in the landing pages are:
`1.0`:
* libraries
  * fluid-framework
  * counter
* service-clients
  * azure-client
  * tinylicious
* utilities
  * azure-service-utils

`2.0`:
* libraries
  * fluid-framework
* service-clients
  * azure-client
  * odsp-client
* utilities
  * azure-service-utils

File: docs/api-markdown-documenter/render-api-documentation.js (hunk 12 above)
Comment (chentong7) [reply docs/api-markdown-documenter/render-api-documentation.js:29]: Right. But it will fail the build with this change only... Should I include the version number change in this PR?

File: docs/api-markdown-documenter/index.js (hunk 9 above)
Comment (chentong7) [reply docs/api-markdown-documenter/index.js:10]: Added

File: docs/api-markdown-documenter/render-api-documentation.js (hunk 13 above)
Comment (chentong7) [reply docs/api-markdown-documenter/render-api-documentation.js:26]: Updated

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 16 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html:92]: If we won't have version number like "1.0.1" then we can use `contains`. If we will then we need the exact match.

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 15 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html:27]: Do we turn on `1.0` only now? should we have `previousVersion` here?

File: docs/data/versions.json (hunk 14 above)
Comment (chentong7) [reply docs/data/versions.json]: Updated.

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 15 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html]: Awesome. Updated

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 17 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html:91]: The easiest way to check the current viewing version is check the url, which includes the version number in. Adding comments here.

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 16 above)
Comment (Josmithr) [reply docs/themes/thxvscode/layouts/partials/apiNav.html:94]: Oh, I see. For the purpose of this system, when we release package versions `2.0.1`, we'll publish their artifacts as `2.0.0-latest`, rather than publishing separate artifacts for each version.

Also note that we will eventually publish versions like `2.1.0`, which would fail this check.

Comment (Josmithr) [reply docs/themes/thxvscode/layouts/partials/apiNav.html:94]: But this highlights the fact that our versioned artifact naming is a bit weird. It might make more sense for us to name our versions here something like `1.x` and `2.x` instead of `1.0` and `2.0`. Or even just `v1` and `v2`. We should discuss this during the next docs sync meeting.

File: docs/api-markdown-documenter/render-api-documentation.js (hunk 12 above)
Comment (Josmithr) [reply docs/api-markdown-documenter/render-api-documentation.js:31]: Oh, I see. I think the way we fix that is by modifying `versions.json` before merging to only include 1.0 docs.

As it is, this will always generate both 1.0 and 2.0 docs, but both versions published to the website will actually be the 2.0 docs, which we don't want.

File: docs/api-markdown-documenter/index.js [hunk 18]
```diff
@@ -4,16 +4,41 @@
...
-	(error) => {
-		console.error("API docs could not be written due to an error:", error);
-		process.exit(1);
-	},
-);
+/**
+ * This is the enable/disable flag for rendering multiple versions of the API docs. (now: false)
+ * To enable this feature, run the following command: build:api-rollup:multi-version after
+ * https://github.com/microsoft/FluidFramework/pull/18677 merged.
+ **/
```
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:14]:
```suggestion
 */
```

File: docs/api-markdown-documenter/index.js [hunk 19]
```diff
@@ -4,16 +4,41 @@
...
-		console.log(chalk.green("API docs written!"));
-		process.exit(0);
-	},
-	(error) => {
-		console.error("API docs could not be written due to an error:", error);
-		process.exit(1);
-	},
-);
+/**
+ * This is the enable/disable flag for rendering multiple versions of the API docs. (now: false)
```
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:11]: This comment is attached to the `renderApiAllDocumentation` function, but it seems to be describing the argument.

I think we should re-work this to document what the whole script does, and what its expected parameters are. That way anyone who wants to know how to invoke it via something like `node api-markdown-documenter true` knows what it does and how to call it.

File: docs/api-markdown-documenter/index.js [hunk 20]
```diff
@@ -4,16 +4,41 @@
...
+ * To enable this feature, run the following command: build:api-rollup:multi-version after
+ * https://github.com/microsoft/FluidFramework/pull/18677 merged.
+ **/
+async function renderApiAllDocumentation() {
```
Comment (Josmithr) [inline docs/api-markdown-documenter/index.js:15]: I would recommend not wrapping this logic in an outer function. As it is, the function is being called at the bottom of the script, but not being awaited. In order for callers to know when this has finished and what the result status was, you would need to await the function call at the bottom of the file. But at that point, wrapping the logic in a function doesn't really offer much benefit.

File: docs/data/versions.json [hunk 21]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (Josmithr) [inline docs/data/versions.json:1]: This format looks really nice! I like this a lot!

Just leaving this comment as a reminder: we'll want to remove all of the 2.0-related entries here before merging. At some point in the future, when we're ready to publish our 2.0 docs, we can re-add those members and everything should automatically work.

File: docs/data/versions.json [hunk 22]
```diff
@@ -0,0 +1,14 @@
...
+	"params": {
+		"currentVersion": "2.0",
+		"ltsVersion": "1.0",
+		"previousVersions": ["1.0"],
+		"allVersions": ["1.0", "2.0"]
```
Comment (Josmithr) [inline docs/data/versions.json:6]: I don't _think_ `allVersions` should be necessary. This will always be `previousVersions` + `currentVersion`, right?

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 17 above)
Comment (Josmithr) [reply docs/themes/thxvscode/layouts/partials/apiNav.html:91]: If possible, it would still be nice to create a single helper function that all of the places can call. But if that's not possible, docs would be great.

**Comment by chentong7** [comment]
Add:
```
"2.0": {
		"libraries": ["fluid-framework", "@fluidframework/counter"],
		"serviceClients": ["@fluidframework/azure-client", "@fluid-experimental/odsp-client"],
		"utilities": ["@fluidframework/azure-service-utils"]
	}
```
into pacakges.json after turning on 2.0

File: docs/data/versions.json (hunk 22 above)
Comment (chentong7) [reply docs/data/versions.json:6]: Removed

File: docs/data/versions.json (hunk 21 above)
Comment (chentong7) [reply docs/data/versions.json:1]: Removed

File: docs/data/versions.json [hunk 23]
```diff
@@ -1,14 +1,12 @@
 {
 	"params": {
-		"currentVersion": "2.0",
+		"currentVersion": "1.0",
 		"ltsVersion": "1.0",
-		"previousVersions": ["1.0"],
-		"allVersions": ["1.0", "2.0"]
+		"previousVersions": ["1.0"]
```
Comment (Josmithr) [inline docs/data/versions.json:5]: This should be empty while 1.0 is current, right?

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 16 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html:108]: Marking the following discussion here.

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 17 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html]: Seems like it's better to have a version matching helper function, then if there is version number naming rule change, it will only need to update one place..

File: docs/api-markdown-documenter/render-api-documentation.js (hunk 12 above)
Comment (chentong7) [reply docs/api-markdown-documenter/render-api-documentation.js:31]: Update this file will break current PR build. Marking down to update this place after https://github.com/microsoft/FluidFramework/pull/18677 merged.

**Comment by chentong7** [comment]
Working on:
1. current/selected api version matching helper function in hugo.
2. Redirect (docs/apis and docs/apis/lts)-> (docs/apis/<currentVersion>) function in hugo.

File: docs/themes/thxvscode/layouts/partials/apiNav.html (hunk 16 above)
Comment (chentong7) [reply docs/themes/thxvscode/layouts/partials/apiNav.html:108]: Discussed in meeting. Name to v1, v2 now

File: docs/api-markdown-documenter/render-api-documentation.js (hunk 12 above)
Comment (chentong7) [reply docs/api-markdown-documenter/render-api-documentation.js]: Merged from main. Updated.

File: docs/.gitignore [hunk 24]
```diff
@@ -1,16 +1,16 @@
...
 node_modules/
 public/
 scripts/api-documenter/temp
 themes/thxvscode/assets/js/node_modules/
 
 # Generated by build-api-nav during docs build.
-data/allAPIs.yaml
-data/displayNameToPackageName.yaml
-data/packageNameToDisplayName.yaml
+data/*/allAPIs.json
```
Comment (RishhiB) [inline docs/.gitignore:11]: The original idea was to put packages.json inside these v1, v2 folders, so I did this so we dont delete packages.json within v1 or v2, since we didnt continue with this design change, we can remove these three lines for a generic `data/v*/`

Comment (chentong7) [reply docs/.gitignore]: Make sense. Updated!

**Review by rohandubal — APPROVED** [review]

File: packages/tools/devtools/devtools-core/README.md [hunk 25]
```diff
@@ -1,12 +1,8 @@
-# @fluidframework/devtools-core
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-core/README.md:1]: @chentong7 it looks like a handful of the package READMEs had their contents replaced with the contents of the `cell` package's README. We'll need to revert these changes.

Comment (Josmithr) [reply packages/tools/devtools/devtools-core/README.md:1]: Looks like the affected packages include:
* counter
* devtools
* devtools-core
* fluid-framework
